tests/Unit/Utilities/PregTest.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Utilities;
+
+use Pelago\Emogrifier\Utilities\Preg;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Pelago\Emogrifier\Utilities\Preg
+ */
+final class PregTest extends TestCase
+{
+    /**
+     * @var string|array<string>
+     */
+    private $replaceCallbackReplacement;
+
+    /**
+     * @var ?string
+     */
+    private $lastReplaceCallbackMatch;
+
+    /**
+     * @var int
+     */
+    private $replaceCallbackReplacementIndex;
```

Comment:
As this is non-nullable, let's initialize it with an int:

```suggestion
    private $replaceCallbackReplacementIndex = 0;
```

Reply:
Done.